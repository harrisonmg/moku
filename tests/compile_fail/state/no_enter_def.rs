use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Bottom {
        something: u8,
    }

    #[superstate(Top)]
    impl State<BlinkyState> for Bottom {}
}

fn main() {}
