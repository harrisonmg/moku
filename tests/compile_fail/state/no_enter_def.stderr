error: a struct with fields must manually implement the `State::enter` function
  --> tests/compile_fail/state/no_enter_def.rs:18:10
   |
18 |     impl State<BlinkyState> for Bottom {}
   |          ^^^^^
